### ISSUE TITLE
`sst-env.d.ts` Generation Missing Export in Monorepo Workspace

### ISSUE THREAD

**Issue by isaac-scarrott** [sst/ion#556]
Hey,

I'm seeing an issue with `sst-env.d.ts` in my monorepo. I have two `sst-env.d.ts` files being generated, one in the root and one in my cloudflare worker folder/workspace.

The root `sst-env.d.ts` seems to be correct and look something like the following:

```ts
/* tslint:disable */
/* eslint-disable */
import "sst"
declare module "sst" {
  export interface Resource {
    Api: {
      type: "sst.cloudflare.Worker"
      url: string
    }
  }
}
export {}
```

However, the `sst-env.d.ts` generated in my app/workspace folder seems to be causing an issue as it does not have export {} at the end and is causing the error `'Resource' only refers to a type, but is being used as a value here when attempting to reference the resource`

Also note that this is a project deployed to cloudflare.

**Comment by isaac-scarrott** [comment]
Please let me know if you wish me to create a minimal repository to reproduce this issue

**Comment by jayair** [comment]
We made some changes to this recently, can you update your CLI?

Also have a look at how our monorepo example has it: https://github.com/sst/ion/blob/dev/examples/aws-monorepo/packages/functions/sst-env.d.ts & https://github.com/sst/ion/blob/dev/examples/aws-monorepo/sst-env.d.ts

Lmk if that helps.

**Comment by isaac-scarrott** [comment]
Hey @jayair, upgrading didn't help unfortunately and I'm following the examples correct. I believe it's [this commit  and specifically this line](https://github.com/sst/ion/commit/937c436a284c98c79000809bc447759f6c2401b1#diff-0a9b9fffa130b1d91db05af90f8cadaf19a1e5134f2c72e7b88287d170e17759L478) that has caused this to happen. 

I believe the fix would be to add `export {}`, to the [line after this one](https://github.com/sst/ion/blob/40043617d1caf20860bd161a44648f9b9187850b/pkg/project/stack.go#L493) would fix this issue. Similar to how there is this line [here](https://github.com/sst/ion/blob/40043617d1caf20860bd161a44648f9b9187850b/pkg/project/stack.go#L456) for the root `sst-env.d.ts` that enables all of my non cloudflare resources to pass the type checks.

I would test this locally and raise a PR however, I can't work out how to run the CLI locally so if I could get some guidance on how to run it locally or someone else could fix it that would be great

**Comment by kleenkanteen** [comment]
I get the same error but when following https://ion.sst.dev/docs/start/aws/hono/. I am on a github codespace with 8gb ram.

**Comment by kleenkanteen** [comment]
I was able to bypass these issues by installing a 2 month old version of ion and the sst npm package.

I did these steps:
1. `rm -rf .ssh` in the current directory
2. ion cli linux install: `curl -fsSL https://ion.sst.dev/install | VERSION=0.0.337 bash`
3. `pnpm install sst@3.0.19`

**Comment by kleenkanteen** [comment]
Ok I stopped `sst dev` on accident, then reran it, and I get this cryptic error, different than the original error I was getting like OP:

`@kleenkanteen ➜ /workspaces/YT-Sub-Steroids/backend (sst-backend) $ sst dev --verbose
time=2024-07-09T06:42:48.811Z level=INFO msg="checking for pulumi" path=/home/codespace/.config/sst/bin/pulumi
time=2024-07-09T06:42:49.442Z level=INFO msg="checking for bun" path=/home/codespace/.config/sst/bin/bun
time=2024-07-09T06:42:49.461Z level=INFO msg=args args=[] length=0
time=2024-07-09T06:42:49.464Z level=INFO msg="no existing server found, starting new one"
time=2024-07-09T06:42:49.473Z level=INFO msg="waiting for server to start"
time=2024-07-09T06:42:49.575Z level=INFO msg="checking for pulumi" path=/home/codespace/.config/sst/bin/pulumi
time=2024-07-09T06:42:50.266Z level=INFO msg="checking for bun" path=/home/codespace/.config/sst/bin/bun
time=2024-07-09T06:42:50.294Z level=INFO msg="initializing project" version=0.0.337
time=2024-07-09T06:42:50.294Z level=INFO msg="esbuild building"
time=2024-07-09T06:42:51.167Z level=INFO msg="esbuild built" outfile=/workspaces/YT-Sub-Steroids/backend/.sst/eval/eval-1720507370294.mjs
time=2024-07-09T06:42:51.167Z level=INFO msg="evaluating config"
time=2024-07-09T06:42:51.440Z level=INFO msg="config evaluated"
time=2024-07-09T06:42:51.440Z level=ERROR msg="exited with error" err="Error evaluating config: exit status 1\nfile:///workspaces/YT-Sub-Steroids/backend/.sst/eval/eval-1720507370294.mjs:33\nvar bucket = new sst.aws.Bucket(\"MyBucket\", {\n             ^\n\nReferenceError: sst is not defined\n    at file:///workspaces/YT-Sub-Steroids/backend/.sst/eval/eval-1720507370294.mjs:33:14\n    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)\n    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)\n    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)\n\nNode.js v20.14.0\n"
×  Unexpected error occurred. Please check the logs or run with --verbose for more details.
time=2024-07-09T06:42:51.442Z level=ERROR msg="exited with error" err="exit status 1"`

**Comment by kleenkanteen** [comment]
Ok somehow moving this part to inside the run() function solved my issue: 
```
const bucket = new sst.aws.Bucket("MyBucket", {
        public: true
        });            
```

**Comment by thdxr** [comment]
@isaac-scarrott sorry having a bit of trouble following

can you show me a full example of what is being generated?

**Comment by thdxr** [comment]
gotcha - should be fixed in v0.0.540

**Comment by isaac-scarrott** [comment]
Thanks @thdxr !
